Linux bootstrap is missing libclang-dev and clang-format, breaking script/presubmit

## Summary

Two packages required by `script/presubmit` on Linux are not installed by `script/linux/install_build_deps` (or any of its callers — `install_runtime_deps`, `install_test_deps`, `bootstrap`). A contributor running `script/linux/bootstrap` on a fresh Ubuntu/Debian machine and then running `script/presubmit` will hit two distinct failures.

## Problem 1: missing `libclang-dev` → `cargo nextest` fails to build `minimp4-sys`

`script/presubmit:56` runs `cargo nextest run --no-fail-fast --workspace`. That activates the `integration_tests` feature on `warpui_core` (`crates/warpui_core/Cargo.toml:12`), which pulls in `minimp4` → `minimp4-sys`. The `minimp4-sys` build script uses `bindgen` to parse C headers, and bindgen calls into `libclang` to do that.

Without `libclang-dev` installed, the build script fails:

```
error: failed to run custom build command for `minimp4-sys v0.1.1`
...
--- stderr
/usr/include/stdio.h:34:10: fatal error: 'stddef.h' file not found

thread 'main' (...) panicked at .../minimp4-sys-0.1.1/build.rs:14:10:
Unable to generate bindings: ClangDiagnostic("...'stddef.h' file not found\n")
```

The missing `stddef.h` lives in clang's resource directory (`/usr/lib/llvm-*/lib/clang/*/include/`). It's shipped by `libclang-common-*-dev`, which is pulled in transitively by `libclang-dev`:

```
libclang-dev → libclang-18-dev → libclang-common-18-dev   (the resource-dir headers)
                               → libclang1-18              (the .so bindgen loads)
```

Note: a plain `cargo build` does **not** trigger this — only test/integration builds do — which is why this hasn't shown up before for contributors who only build but don't run presubmit locally.

## Problem 2: missing `clang-format` → presubmit's clang-format check fails

`script/presubmit:32` calls `./script/run-clang-format.py -r --extensions 'c,h,cpp,m' ./crates/warpui/src/ ./app/src/`, which requires the `clang-format` binary on PATH.

`clang-format` is installed on macOS by `script/macos/bootstrap:44` via Homebrew, but is **never installed** by any of the Linux scripts. So presubmit fails for any Linux contributor who hasn't installed it manually.

## Reproduction

On a fresh Ubuntu 24.04 (or any Debian-derived) machine:

```bash
git clone https://github.com/warpdotdev/warp
cd warp
script/linux/bootstrap
script/presubmit
```

Expected: presubmit passes.
Actual: fails on `clang-format` (step at line 32) or, after `clang-format` is installed manually, fails on `cargo nextest` building `minimp4-sys`.

## Suggested fix

Add both packages to the `PACKAGES=( ... )` array in `script/linux/install_build_deps` (lines 18–36):

```bash
# Needed by bindgen (used by minimp4-sys, pulled in by warpui_core's
# integration_tests feature) — pulls in libclang-common-*-dev which
# provides clang's resource-dir builtin headers.
libclang-dev
# Required by script/presubmit's clang-format check on C/C++/Obj-C sources.
clang-format
```

Both fit `install_build_deps` (compile-/build-time tooling) rather than `install_test_deps` (runtime tools the test suite spawns as subprocesses, e.g. zsh/fish/vim/gcloud). Precedent: `libasound2-dev` is in build_deps despite being only needed for the voice-input feature.

## Environment

- OS: Ubuntu 24.04 (noble)
- Pre-installed: `libclang1-18`, `llvm-18` (as transitive deps of other packages)
- Not pre-installed: `libclang-dev`, `clang-format`

--- a/script/linux/install_build_deps
+++ b/script/linux/install_build_deps
@@ -33,6 +33,12 @@ if [[ "$(source /etc/os-release; echo $ID $ID_LIKE)" = *"debian"* ]]; then
     brotli
     # Needed for voice input
     libasound2-dev
+    # Needed by bindgen (used by minimp4-sys, pulled in by warpui_core's
+    # integration_tests feature) — pulls in libclang-common-*-dev which
+    # provides clang's resource-dir builtin headers.
+    libclang-dev
+    # Required by script/presubmit's clang-format check on C/C++/Obj-C sources.
+    clang-format
   )
   sudo apt-get install -y "${PACKAGES[@]}"
 

